include/esp-stub-lib/uart.h

```diff
@@ -0,0 +1,148 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ */
+
+#pragma once
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <soc/soc_caps.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif // __cplusplus
+
+/**
+ * @brief UART port number type
+ *
+ * Available UART ports depend on the chip's SOC_UART_HP_NUM capability.
+ * Use the UART_NUM_* enum values (e.g., UART_NUM_0, UART_NUM_1, etc.)
+ */
+
+typedef enum {
+    UART_NUM_0,                         /*!< UART port 0 */
+    UART_NUM_1,                         /*!< UART port 1 */
+#if SOC_UART_HP_NUM > 2
+    UART_NUM_2,                         /*!< UART port 2 */
+#endif
+#if SOC_UART_HP_NUM > 3
+    UART_NUM_3,                         /*!< UART port 3 */
+#endif
+#if SOC_UART_HP_NUM > 4
+    UART_NUM_4,                         /*!< UART port 4 */
+#endif
+    UART_NUM_MAX,                       /*!< UART port max */
+} uart_port_t;
+
+#define UART_INTR_RXFIFO_FULL      (0x1 << 0)
+#define UART_INTR_RXFIFO_TOUT      (0x1 << 8)
+
+/**
+ * @brief Wait until UART is idle
+ *
+ * @param uart_num UART port number
+ */
+void stub_lib_uart_wait_idle(uint8_t uart_num);
```

Comment:
Since we have a new type for uart_num and it is public, we can use it in the parameter.
```suggestion
void stub_lib_uart_wait_idle(uart_port_t uart_num);
```